code/_onclick/click_alt.dm

```diff
@@ -6,6 +6,11 @@
 /mob/proc/base_click_alt(atom/target)
 	SHOULD_NOT_OVERRIDE(TRUE)
 
+	if(HAS_TRAIT(src, TRAIT_MECH_PILOT) && istype(loc, /obj/vehicle/sealed/mecha))
+		if(target == loc)
+			target.click_alt(src)
+		return
+
 	var/turf/tile = isturf(target) ? target : get_turf(target)
```

Comment:
Unrelated but `get_turf(of_a_turf)` works, you can save yourself some boilerplate here